src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1367,8 +1367,8 @@ FileIdentifier getFileIdentifierForFileTypes()
                 {
                     if (binder is BuckStopsHereBinder lastBinder)
                     {
-                        // we never expect to bind a file type in a context where the BuckStopsHereBinder lacks an AssociatedFileIdentifier
-                        return lastBinder.AssociatedFileIdentifier.Value;
+                        // BuckStopsHereBinder.AssociatedFileIdentifier may be null from the EE.
+                        return lastBinder.AssociatedFileIdentifier.GetValueOrDefault();
```

Comment:
I guess if the file identifier we get for this binder chain is default, we simply return false from `IsInScopeOfAssociatedSyntaxTree`?

Reply:
Consider asserting `lastBinder.AssociatedFileIdentifier.HasValue || (this.Flags & BinderFlags.InEEMethodBinder) != 0`.

Reply:
> I guess if the file identifier we get for this binder chain is default, we simply return false from `IsInScopeOfAssociatedSyntaxTree`?

Correct.

> Consider asserting `lastBinder.AssociatedFileIdentifier.HasValue || (this.Flags & BinderFlags.InEEMethodBinder) != 0`.

Sounds good. I'll add that in a subsequent PR unless other changes are needed here.